pyronear/datasets/openfire.py

```diff
@@ -42,13 +54,16 @@ class OpenFire(VisionDataset):
     seed = 42
 
     def __init__(self, root, train=True, transform=None, target_transform=None,
-                 download=False, threads=16, valid_pct=None):
+                 download=False, threads=16, valid_pct=0.2, max_files=-1):
         super(OpenFire, self).__init__(root, transform=transform,
                                     target_transform=target_transform)
         self.train = train  # training set or test set
-
-        if download:
-            self.download(threads, valid_pct)
+        
+        if not (self._check_exists(train=True) and self._check_exists(train=False)):
```

Comment:
Not sure to understand the logic behind this line.
The lines below will be executed only if test set exists already but not the training one?

Reply:
They will be skipped if both exist. Maybe you missed the () ?
(N.B.: this will disappear if we create the .pt files each time)

Reply:
Indeed, my bad. Better to go for  `not A or not B` (it will stop earlier if the first condition is met, whereas in your suggestion both will be evaluated)

Reply:
Aesthetically I agree but I believe both are equivalent. The following does not execute the 2nd statement:

```
import time
if not (False and time.sleep(5)): pass
```

Reply:
I agree but I was referring to execution time, but I just tested it and it doesn't seem to be doing much difference indeed so that's alright!